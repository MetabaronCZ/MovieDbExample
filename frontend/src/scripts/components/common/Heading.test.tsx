import { getByText } from '@testing-library/dom';
import { describe, expect, it } from '@jest/globals';

import { Heading } from './Heading';
import { TestComponentWrapper, waitForComponent } from 'test-utils/component';

describe('components/common/Heading', () => {
  it('should render properly', async () => {
    const { container } = await waitForComponent(<Heading>CONTENT</Heading>, {
      wrapper: TestComponentWrapper,
    });
    expect(container).toMatchSnapshot();

    const heading = getByText(container, 'CONTENT');
    expect(heading).toBeInTheDocument();
  });

  it('should render different variations', async () => {
    const { container: h1 } = await waitForComponent(
      <Heading tag="h1">HEADING_H1</Heading>,
      { wrapper: TestComponentWrapper },
    );
    expect(h1).toMatchSnapshot();
    expect(h1.querySelector('h1')).toBeInTheDocument();

    const { container: h2 } = await waitForComponent(
      <Heading tag="h2">HEADING_H2</Heading>,
      { wrapper: TestComponentWrapper },
    );
    expect(h2).toMatchSnapshot();
    expect(h2.querySelector('h2')).toBeInTheDocument();

    const { container: h3 } = await waitForComponent(
      <Heading tag="h3">HEADING_H3</Heading>,
      { wrapper: TestComponentWrapper },
    );
    expect(h3).toMatchSnapshot();
    expect(h3.querySelector('h3')).toBeInTheDocument();

    const { container: h4 } = await waitForComponent(
      <Heading tag="h4">HEADING_H4</Heading>,
      { wrapper: TestComponentWrapper },
    );
    expect(h4).toMatchSnapshot();
    expect(h4.querySelector('h4')).toBeInTheDocument();
  });
});
