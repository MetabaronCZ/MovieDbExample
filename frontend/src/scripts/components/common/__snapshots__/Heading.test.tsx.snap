// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/common/Heading should render different variations 1`] = `
.c0 {
  font-family: Tahoma,Verdana,Arial,sans-serif;
  font-size: 14px;
  line-height: 24px;
  color: #002d56;
  font-size: 18px;
  line-height: 24px;
  font-weight: 400;
}

<div>
  <h1
    class="c0"
  >
    HEADING_H1
  </h1>
</div>
`;

exports[`components/common/Heading should render different variations 2`] = `
.c0 {
  font-family: Tahoma,Verdana,Arial,sans-serif;
  font-size: 14px;
  line-height: 24px;
  color: #002d56;
  font-size: 18px;
  line-height: 24px;
  font-weight: 400;
}

<div>
  <h2
    class="c0"
  >
    HEADING_H2
  </h2>
</div>
`;

exports[`components/common/Heading should render different variations 3`] = `
.c0 {
  font-family: Tahoma,Verdana,Arial,sans-serif;
  font-size: 14px;
  line-height: 24px;
  color: #002d56;
  font-size: 18px;
  line-height: 24px;
  font-weight: 400;
}

<div>
  <h3
    class="c0"
  >
    HEADING_H3
  </h3>
</div>
`;

exports[`components/common/Heading should render different variations 4`] = `
.c0 {
  font-family: Tahoma,Verdana,Arial,sans-serif;
  font-size: 14px;
  line-height: 24px;
  color: #002d56;
  font-size: 18px;
  line-height: 24px;
  font-weight: 400;
}

<div>
  <h4
    class="c0"
  >
    HEADING_H4
  </h4>
</div>
`;

exports[`components/common/Heading should render properly 1`] = `
.c0 {
  font-family: Tahoma,Verdana,Arial,sans-serif;
  font-size: 14px;
  line-height: 24px;
  color: #002d56;
  font-size: 18px;
  line-height: 24px;
  font-weight: 400;
}

<div>
  <h2
    class="c0"
  >
    CONTENT
  </h2>
</div>
`;
